src/tir/transforms/lower_intrin.cc

```diff
@@ -112,20 +113,33 @@ class IntrinInjecter : public tvm::arith::IRMutatorWithAnalyzer {
       // Common path, positive divisor
       if (analyzer_->CanProveGreaterEqual(op->a, 0) || analyzer_->CanProveGreaterEqual(e, 0)) {
         return truncdiv(op->a, op->b);
+      }
+
+      // If the numerator's lower bound is known, express the floordiv
+      // in terms of truncdiv using only positive operands.
+      arith::ConstIntBound const_int_bound = analyzer_->const_int_bound(op->a);
+      if (const_int_bound->min_value != arith::ConstIntBound::kNegInf &&
+          const_int_bound->min_value < 0 &&
+          const_int_bound->min_value > -(1LL << (op->a->dtype.bits() - 1))) {
```

Comment:
Thank you, and op.h's `min_value` is exactly what I had been looking for.  I had checked whether `ConstIntBound::Everything(dtype)` was public, but wasn't aware of `min_value`.  PR updated to use `min_value` instead of doing the calculation here.